### PR TITLE
Restores collector and editor logic in view. Fixes #1477

### LINKED ISSUE #1477: Collector and editor roles in view
View `app/views/curation_concerns/base/_show_actions.html.erb` depends on two variables (`collector` and `editor`) that are not defined. There is a TODO in the view at the moment indicating that this needs to be fixed (see PR #1476). It's possible that this code will be replaced with calls to `ability`

### PR DESCRIPTION
**Pull request by hectorcorrea**

### CODE REVIEW

File: app/presenters/sufia/work_show_presenter.rb [hunk 1]
```diff
@@ -3,6 +3,14 @@ class WorkShowPresenter < ::CurationConcerns::WorkShowPresenter
     # delegate fields from Sufia::Works::Metadata to solr_document
     delegate :based_near, :depositor, :identifier, :resource_type, :tag, to: :solr_document
 
+    def collector
+      current_ability.can?(:collect, solr_document)
+    end
+
+    def editor
```
Comment (grosscol) [inline app/presenters/sufia/work_show_presenter.rb]: Should this be `editor?` if it's returning true|false

File: app/presenters/sufia/work_show_presenter.rb [hunk 2]
```diff
@@ -3,6 +3,14 @@ class WorkShowPresenter < ::CurationConcerns::WorkShowPresenter
     # delegate fields from Sufia::Works::Metadata to solr_document
     delegate :based_near, :depositor, :identifier, :resource_type, :tag, to: :solr_document
 
+    def collector
```
Comment (grosscol) [inline app/presenters/sufia/work_show_presenter.rb]: Should this be `collector?` if it's returning true|false

File: app/views/curation_concerns/base/_show_actions.html.erb [hunk 3]
```diff
@@ -1,8 +1,4 @@
-<%# TODO: set these variables based on the current user
-  # See https://github.com/projecthydra/sufia/issues/1477 %>
-<% collector = true %>
-<% editor = true %>
-<% if collector || editor %>
+<% if @presenter.collector || @presenter.editor %>
```
Comment (grosscol) [inline app/views/curation_concerns/base/_show_actions.html.erb]: Perhaps leave the collector role for another PR as suggested by @mjgiarlo.  Permitting analytics and citations to be viewable by anyone with permissions to view the object seems reasonable to me.

File: app/presenters/sufia/file_set_presenter.rb [hunk 4]
```diff
@@ -4,6 +4,14 @@ class FileSetPresenter < ::CurationConcerns::FileSetPresenter
 
     delegate :depositor, :tag, :date_created, :date_modified, to: :solr_document
 
+    def collector
```
Comment (mjgiarlo) [inline app/presenters/sufia/file_set_presenter.rb]: Can you remove all references to `collector` in this PR? It's a role that exists in CC but hasn't existed in Sufia, so I'd like to revisit roles more broadly in a future PR.

File: app/presenters/sufia/file_set_presenter.rb [hunk 5]
```diff
@@ -4,6 +4,14 @@ class FileSetPresenter < ::CurationConcerns::FileSetPresenter
...
+      current_ability.can?(:collect, solr_document)
+    end
+
+    def editor
```
Comment (mjgiarlo) [inline app/presenters/sufia/file_set_presenter.rb]: Could you rename to `editor?` to reflect that this is a boolean method?

File: app/presenters/sufia/work_show_presenter.rb (hunk 2 above)
Comment (mjgiarlo) [inline app/presenters/sufia/work_show_presenter.rb]: Remove this method, pls.

File: app/presenters/sufia/file_set_presenter.rb (hunk 4 above)
Comment (grosscol) [inline app/presenters/sufia/file_set_presenter.rb]: Push the discussion of collector to a separate ticket regarding roles and role management?

File: app/views/curation_concerns/base/_show_actions.html.erb (hunk 3 above)
Comment (mjgiarlo) [inline app/views/curation_concerns/base/_show_actions.html.erb]: I don't think we want to restrict viewing analytics or citations to any particular roles. Remove this line.

File: app/views/curation_concerns/base/_show_actions.html.erb [hunk 6]
```diff
@@ -10,7 +6,7 @@
     <% if Sufia.config.citations %>
       <%= link_to "Citations", sufia.citations_work_path(@presenter), id: 'citations', class: 'btn btn-default' %>
     <% end %>
-    <% if editor %>
+    <% if @presenter.editor %>
```
Comment (mjgiarlo) [inline app/views/curation_concerns/base/_show_actions.html.erb]: `s/editor/editor?/`

**Comment by mjgiarlo** [comment]
Ha, turns out @grosscol and I were leaving the same comments at the same time. Sorry for the dogpile, @hectorcorrea. :smile:

**Comment by hectorcorrea** [comment]
@grosscol @mjgiarlo Done. Removed the `collector` logic and renamed `editor` to `editor?`

**Comment by mjgiarlo** [comment]
:+1:
